models/gif_psc_exp.h

```diff
@@ -0,0 +1,425 @@
+   * Hiding
+   */
+  using Node::handle;
+  using Node::handles_test_event;
+
+  port send_test_event( Node&, rport, synindex, bool );
+
+  void handle( SpikeEvent& );
+  void handle( CurrentEvent& );
+  void handle( DataLoggingRequest& );
+
+  port handles_test_event( SpikeEvent&, rport );
+  port handles_test_event( CurrentEvent&, rport );
+  port handles_test_event( DataLoggingRequest&, rport );
+
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+private:
+  void init_state_( const Node& proto );
+  void init_buffers_();
+  void calibrate();
+
+  void update( Time const&, const long_t, const long_t );
+
+  // The next two classes need to be friends to access the State_ class/member
+  friend class RecordablesMap< gif_psc_exp >;
+  friend class UniversalDataLogger< gif_psc_exp >;
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Independent parameters of the model.
+   */
+  struct Parameters_
+  {
+
+    double_t g_L_;
+    double_t E_L_;
+    double_t V_reset_;
+    double_t delta_u_;
+    double_t v_t_star_;
+    double_t lambda0_;
+
+    /** Refractory period in ms. */
+    double_t t_ref_;
+
+    /** Membrane capacitance in pF. */
+    double_t c_m_;
+
+    /** List of spike triggered current time constant in ms. */
+    std::vector< double_t > tau_stc_;
+
+    /** List of spike triggered current jumps in nA. */
+    std::vector< double_t > q_stc_;
+
+    /** List of adaptive threshold time constant in ms. */
+    std::vector< double_t > tau_sfa_;
+
+    /** List of adaptive threshold jumps in mV. */
+    std::vector< double_t > q_sfa_;
+
+    /** Time constant of excitatory synaptic current in ms. */
+    double_t tau_ex_;
+
+    /** Time constant of inhibitory synaptic current in ms. */
+    double_t tau_in_;
+
+
+    /** External DC current. */
+    double_t I_e_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    double_t y0_;  //!< This is piecewise constant external current
+    double_t y3_;  //!< This is the membrane potential RELATIVE TO RESTING POTENTIAL.
+    double_t q_;   //!< This is the change of the 'threshold' due to adaptation.
+    double_t stc_; // Spike triggered current.
+
+    std::vector< double_t > q_sfa_elems_; // Vector of adaptation parameters.
+    std::vector< double_t > q_stc_elems_; // Vector of spike triggered parameters.
+
+    double_t i_syn_ex_; // postsynaptic current for exc.
+    double_t i_syn_in_; // postsynaptic current for inh.
+
+    int_t r_ref_; // absolute refractory counter (no membrane potential propagation)
+
+    bool initialized_; // it is true if the vectors are initialized
```

Comment:
The name should be more specific. e.g., `sfa_stc_initialized_`, but it might be unnecessary, see comments in cpp-file.